### PR TITLE
Multiprocess RQ workers (using supervisor)

### PR DESCRIPTION
**Pull request by rauchy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
This is an alternative implementation to the one suggested in #4233. Here we use `supervisor` instead of `Honcho` in order to spin up, monitor and restart multiple worker processes.

supervisor is installed in the Docker image and is launched inside the `worker` entrypoint. We use the non-daemonized mode of supervisor in order to allow the container to exit in case the supervisor process exits.

## Related Tickets & Documents
#4233

### CODE REVIEW

File: bin/docker-entrypoint [hunk 1]
```diff
@@ -25,13 +25,22 @@ dev_scheduler() {
...
-  exec watchmedo auto-restart --directory=./redash/ --pattern=*.py --recursive -- ./manage.py rq worker $QUEUES
+  start_worker "watchmedo auto-restart --directory=./redash/ --pattern=*.py --recursive -- "
+}
+
+start_worker() {
+  export EXECUTION_PREFIX=$1
+  export WORKERS_COUNT=${WORKERS_COUNT:-2}
+  export QUEUES=${QUEUES:-}
+
+  touch /tmp/worker.log && tail -f /tmp/worker.log &
```
Comment (arikfr) [inline bin/docker-entrypoint]: Is there a reason you didn't configure supervisord to write to stdout directly?

Comment (rauchy) [reply bin/docker-entrypoint]: That's even simpler :) didn't think of that! 1fb1a7e handles this.

File: worker.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[supervisord]
+logfile=/tmp/supervisord.log
```
Comment (arikfr) [inline worker.conf]: Need to update this to stdout as well.

Comment (rauchy) [reply worker.conf]: fdbbd13

**Comment by rauchy** [comment]
> We still some healthcheck, right? Can Supervisor handle it?
In 092eb6c I've used [supervisor_checks](https://github.com/vovanec/supervisor_checks) to run a check every minute - the check runs for every monitored process and asserts that the worker is behaving as expected (on top of what you get from supervisor, which is basically testing if the process is up). If a specific worker process is misbehaving, it is replaced.

We can also add more checks for CPU / memory (provided OOTB by supervisor_checks)

Let me know what you think of this.

**Comment by rauchy** [comment]
@arikfr can we merge this?

**Pull request merged**